apple/Sources/Sargon/Drivers/ProfileChange/ProfileChange.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import SargonUniFFI
+import AsyncExtensions
+
+// Makes it possible to type `.shared` on an initalizer/func taking
+// `some EventBusDriver` as parameter.
+extension ProfileChangeDriver where Self == ProfileChangeBus {
+
+    public static var shared: Self { Self.shared }
+}
+
+/// An `EventBusDriver` actor which handles incoming
+/// `EventNotifications` and forwards them to any
+/// subscriber of `notifications()`, being a multicasted
+/// async sequence.
+public final actor ProfileChangeBus {
```

Comment:
`Bus` alone does not make sense IMO. `Event Bus` is a common term, but it is most commonly used when it is a singleton for many different kinds of events, which is not the case here.

Maybe `ProfileChangeNotifier` or `ProfileChangeEventPublisher` ?

Also update incorrect copy paste errors in documentation. And hmm, maybe we want a single shape for this? A generic? Seing as this type is almost identical to `actor EventBus` (which at least in commit 498f627defb421a0f49c495ff68e10b19e77b2ed was not removed ).

and then you can:

```swift
public final actor SargonEventPublisher<Element> { ... }
extension SargonEventPublisher:  ProfileChangeDriver  where Element ==  Profile {
     public func handleProfileChange(changedProfile: Profile) async {
        subject.send(changedProfile)
    }
}

```
and then you can have a single actor?

Reply:
I would rename it, will see about the generic suggestion.

Reply:
if you try the generic approach, I think you will indeed see that they have a 95% identical shape.